frontend/src/metabase/query_builder/actions/native.ts

```diff
@@ -102,7 +102,7 @@ export const setIsShowingSnippetSidebar = (
 
 export const setIsNativeEditorOpen = (isNativeEditorOpen: boolean) => ({
   type: SET_UI_CONTROLS,
-  payload: { isNativeEditorOpen },
+  payload: { isNativeEditorOpen, isShowingDataReference: isNativeEditorOpen },
```

Comment:
The data reference now becomes the integral part of the action that controls the native editor visibility.